doc/boot-messages.md

```diff
@@ -0,0 +1,65 @@
+# Accessing Boot Logs in vfkit
+
+There are different ways to access boot logs in vfkit, depending on the following factors:
+
+## Boot Method
+
+### Direct Kernel Boot (`--bootloader linux`)
+This method requires the user to provide a kernel image and an initrd. You can pass kernel command-line arguments to configure logging output. In the case of `vfkit`, use `console=hvc0` to direct logs to the virtio console.
+
+To capture these logs on a file on the host, use the `--device virtio-serial` with a `logFilePath`. You can refer to [vfkit serial port docs](https://github.com/crc-org/vfkit/blob/main/doc/usage.md#serial-port) for more details on how to use the virtio-serial device.
+
+```
+vfkit --cpus 2 --memory 2048 \
+  --bootloader linux,kernel=/path/to/Image,initrd=/path/to/initrd.img,cmdline="root=/dev/vda1 console=hvc0" \
+  --device virtio-blk,path=/path/to/Fedora-Image.raw \
+  --device virtio-serial,logFilePath=/tmp/vm-console.log
+```
+
+Logs will appear in `/tmp/vm-console.log` on the host.
+
+### EFI Boot (`--bootloader efi`)
+For EFI-booted images (e.g., Fedora Cloud/CoreOS), logs can be redirected by modifying the GRUB configuration inside the VM:
+
+```
+sudo vi /etc/default/grub
+```
+
+Add or update these lines:
+
+```
+GRUB_CMDLINE_LINUX="console=hvc0"
+GRUB_TERMINAL="serial console"
+```
+
+Then regenerate the GRUB config:
+
+```
+sudo grub2-mkconfig -o /boot/grub2/grub.cfg
+```
+
+After this, launch vfkit again with the following options and you can access the logs in `/tmp/vm-console.log` on host:
+
+```
+vfkit --cpus 2 --memory 2048 \
+  --bootloader efi,variable-store=out/efistore.nvram,create \
+  --device virtio-blk,path=/path/to/Fedora-Image.raw \
+  --device virtio-serial,logFilePath=/tmp/vm-console.log
+```
+
+## Image Used
+
+### Fedora CoreOS / Fedora Cloud
+These images support EFI boot and can be modified inside a running VM to change the grub configuration as described above.
+
+### Custom-built Images
+If you have a custom-built image with your own kernel/initrd, the logging must be configured at the kernel cmdline level (`console=hvc0`).
+
+## Console Output Target
+
+You can access boot logs by attaching a virtio-serial device to redirect logs to the host, by enabling the `--gui` option to view logs in a graphical window, or by using both methods simultaneously for parallel access.
+
+1. `--device virtio-serial` can be used with multiple parameters such as `logFilePath`, `stdin`, and `pty` to redirect VM logs to the host. Refer to [vfkit serial port usage](https://github.com/crc-org/vfkit/blob/main/doc/usage.md#serial-port).
+2. `--gui` can be used to open a terminal window to the VM and the logs can be seen on the window. Refer to [vfkit GUI usage](https://github.com/crc-org/vfkit/blob/main/doc/usage.md#enabling-a-graphical-user-interface).
```

Comment:
Are the kernel logs printed to the GUI window with `console=hvc0`? This did not seem to work for me.